packages/website/docs/examples/alert-block.md

```diff
@@ -0,0 +1,464 @@
+
+// Ensures the Alert background color is correct for both light & dark theme
+const componentStyles: Theme["componentStyles"] = (theme) => ({
+  Editor: {
+    ".alert": {
+      backgroundColor: theme.colors.hovered.background,
+      ".alert-icon": {
+        color: theme.colors.hovered.background,
+      },
+    },
+  },
+});
+
+// Adds the theme-dependent Alert styles to the default theme
+const theme = {
+  light: {
+    ...lightDefaultTheme,
+    componentStyles: componentStyles,
+  } satisfies Theme,
+  dark: {
+    ...darkDefaultTheme,
+    componentStyles: componentStyles,
+  } satisfies Theme,
+};
+
+export default function App() {
+  const editor = useBlockNote({
+    blockSchema: schemaWithAlert,
+    slashMenuItems: [
+      ...getDefaultReactSlashMenuItems(schemaWithAlert),
+      insertAlert,
+    ],
+  });
+
+  return (
+    // Theme is applied to the editor
+    <BlockNoteView editor={editor} theme={theme}>
+      {/*Custom Formatting Toolbar - same as default, but the block type*/}
+      {/*dropdown also includes the Alert block*/}
+      <CustomFormattingToolbar editor={editor} />
+      {/*Other toolbars & menus are defaults*/}
+      <HyperlinkToolbarPositioner editor={editor} />
+      <SlashMenuPositioner editor={editor} />
+      <SideMenuPositioner editor={editor} />
+    </BlockNoteView>
+  );
+}
+```
+
+```typescript-vue /Alert.tsx
+import {
+  BlockNoteEditor,
+  BlockSpec,
+  defaultBlockSchema,
+  DefaultBlockSchema,
+  defaultProps,
+  PropSchema,
+  SpecificBlock,
+} from "@blocknote/core";
+import { useState } from "react";
+import {
+  createReactBlockSpec,
+  InlineContent,
+  ReactSlashMenuItem,
+} from "@blocknote/react";
+import { RiAlertFill } from "react-icons/ri";
+import { MdCancel, MdCheckCircle, MdError, MdInfo } from "react-icons/md";
+import { Menu } from "@mantine/core";
+
+// The types of alerts that users can choose from
+export const alertTypes = {
+  warning: {
+    icon: MdError,
+    color: "#e69819",
+  },
+  error: {
+    icon: MdCancel,
+    color: "#d80d0d",
+  },
+  info: {
+    icon: MdInfo,
+    color: "#507aff",
+  },
+  success: {
+    icon: MdCheckCircle,
+    color: "#0bc10b",
+  },
+} as const;
+
+// The props for the Alert block
+export const alertPropSchema = {
+  textAlignment: defaultProps.textAlignment,
+  textColor: defaultProps.textColor,
+  type: {
+    default: "warning" as const,
+    values: ["warning", "error", "info", "success"] as const,
+  },
+} satisfies PropSchema;
+
+// Component for the Alert block
+export const Alert = (props: {
+  block: SpecificBlock<
+    DefaultBlockSchema & { alert: BlockSpec<"alert", typeof alertPropSchema> },
+    "alert"
+  >;
+  editor: BlockNoteEditor<
+    DefaultBlockSchema & { alert: BlockSpec<"alert", typeof alertPropSchema> }
+  >;
+}) => {
+  const [type, setType] = useState(props.block.props.type);
+  const Icon = alertTypes[type].icon;
+
+  const alertStyles = {
```

Comment:
Imo better to leave them in the component, esp since one of them depends on the `type` state

Reply:
Fair, we still have to partially define the styles in the component though, e.g.
```ts
const themedAlertStyles = {
    ...alertStyles,
    backgroundColor: alertTypes[type].backgroundColor[props.theme],
  } as const;
```
Because declaring styles as:
```jsx
styles={{
  ...alertStyles,
  backgroundColor: alertTypes[type].backgroundColor[props.theme],
}}
```
Causes an error - anything in double curly braces will be interpreted as a vue variable.